internal/services/mobilenetwork/mobile_network_attached_data_network_resource_test.go

```diff
@@ -0,0 +1,271 @@
+  user_plane_access_ipv4_gateway           = "10.204.141.1"
+  user_plane_access_ipv4_subnet            = "10.204.141.0/24"
+
+  depends_on = [azurerm_mobile_network_data_network.test]
+}
+`, config, data.Locations.Primary)
+}
+
+func (r MobileNetworkAttachedDataNetworkResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_mobile_network_attached_data_network" "test" {
+  name                                        = "acctest-mnadn-%d"
+  mobile_network_packet_core_data_plane_id    = azurerm_mobile_network_packet_core_data_plane.test.id
+  location                                    = "%s"
+  dns_addresses                               = ["1.1.1.1"]
+  user_equipment_address_pool_prefixes        = ["2.4.1.0/24"]
+  user_equipment_static_address_pool_prefixes = ["2.4.2.0/24"]
+  user_plane_access_name                      = "test"
+  user_plane_access_ipv4_address              = "10.204.141.4"
+  user_plane_access_ipv4_gateway              = "10.204.141.1"
+  user_plane_access_ipv4_subnet               = "10.204.141.0/24"
+
+  network_address_port_translation_configuration {
+    enabled                = true
+    pinhole_maximum_number = 65536
+
+    pinhole_timeouts_in_seconds {
+      icmp = 30
+      tcp  = 100
+      udp  = 39
+    }
+
+    port_range {
+      max_port = 49999
+      min_port = 1024
+    }
+    port_reuse_minimum_hold_time_in_seconds {
+      tcp = 120
+      udp = 60
+    }
+  }
+
+  tags = {
+    key = "value"
+  }
+
+  depends_on = [azurerm_mobile_network_data_network.test]
+
+}
+`, r.template(data), data.RandomInteger, data.Locations.Primary)
+}
+
+func (r MobileNetworkAttachedDataNetworkResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_mobile_network_attached_data_network" "test" {
+  name                                        = "acctest-mnadn-%d"
+  mobile_network_packet_core_data_plane_id    = azurerm_mobile_network_packet_core_data_plane.test.id
+  location                                    = "%s"
+  dns_addresses                               = ["1.1.1.1"]
+  user_equipment_address_pool_prefixes        = ["2.4.3.0/24"]
+  user_equipment_static_address_pool_prefixes = ["2.4.4.0/24"]
+  user_plane_access_name                      = "test"
+  user_plane_access_ipv4_address              = "10.204.141.4"
+  user_plane_access_ipv4_gateway              = "10.204.141.1"
+  user_plane_access_ipv4_subnet               = "10.204.141.0/24"
+
+  network_address_port_translation_configuration {
+    enabled                = true
+    pinhole_maximum_number = 65536
+    pinhole_timeouts_in_seconds {
+      icmp = 30
+      tcp  = 100
+      udp  = 39
+    }
+    port_range {
+      max_port = 49999
+      min_port = 1024
+    }
+    port_reuse_minimum_hold_time_in_seconds {
+      tcp = 120
+      udp = 60
+    }
+  }
+
+  tags = {
+    key = "value"
+  }
+
+  depends_on = [azurerm_mobile_network_data_network.test]
```

Comment:
This (and the same for the other test cases) can be replaced by an implicit dependency on the name of `azurerm_mobile_network_data_network.test`.